src/newton_solver.cpp

```diff
@@ -216,45 +177,38 @@ NewtonSolverDense::NewtonSolverDense(Model *model, ReturnData *rdata,
 /* ----------------------------------------------------------------------------------
  */
 
-int NewtonSolverDense::prepareLinearSystem(int ntry, int nnewt) {
+void NewtonSolverDense::prepareLinearSystem(int ntry, int nnewt) {
     /**
      * Writes the Jacobian for the Newton iteration and passes it to the linear
      * solver
      *
      * @param[in] ntry integer newton_try integer start number of Newton solver
      * (1 or 2)
      * @param[in] nnewt integer number of current Newton step
-     * @return stats integer flag indicating success of the method
      */
 
     /* Get Jacobian */
-    int status = model->fJ(model->nx, tdata->t, 0, tdata->x, tdata->dx,
+    model->fJ(model->nx, tdata->t, 0, tdata->x, tdata->dx,
                            tdata->xdot, tdata->Jtmp, tdata, tmp1, tmp2, tmp3);
-
-    if (status == AMICI_SUCCESS) {
-        /* Compute factorization and pivoting */
-        status = DenseGETRF(tdata->Jtmp, pivots);
-    }
-
-    return status;
+    int status = DenseGETRF(tdata->Jtmp, pivots);
+    if(status != 0)
+        throw NewtonFailure("Dense factorization failed!");
```

Comment:
I like this way to throw errors: Here, you can give much more detailed error messages.